docs/ref/models/indexes.txt

```diff
@@ -0,0 +1,85 @@
+=====================
+Model index reference
+=====================
+
+.. module:: django.db.models.indexes
+   :synopsis: Built-in index types.
+
+.. currentmodule:: django.db.models
+
+.. versionadded:: 1.11
+
+Index classes define a specific database index which can be created through
+them. This document explains the API references of :class:`BaseIndex` which is
+the base class for all other index classes. This includes the `index options`_
+and `built-in indexes`_ Django offers. The indexes can be created in the
+database via the :class:`~django.db.migrations.operations.AddIndex` class
+as follows::
+
+    from django.db import migrations, models
+
+    class Migration(migrations.Migration):
+
+        operations = [
+            migrations.AddIndex(
+                model_name='Book',
+                models.Index('title', 'author', name='my_index_name')  # An index on the title and author fields
```

Comment:
This gives: `SyntaxError: positional argument follows keyword argument`.

After fixing that, I still get a traceback when trying to migrate:
```
File "/home/tim/code/django/django/db/migrations/operations/models.py", line 748, in state_forwards
    model_state.options['indexes'].append(self.index)
KeyError: 'indexes'
```